### ISSUE TITLE
[Table] Crash when undoing row insertion and adding cell properties

### ISSUE THREAD

**Issue by FilipTokarski** [ckeditor/ckeditor5#6382]
## 📝 Provide detailed reproduction steps (if any)

1. Go to `tableselection` test and add a table
2. Select single cell or multiple cells
3. Add row above or below
4. Select the whole new row
5. Open cell properties and add background color
6. Without changing the selection, `undo` two times

### ✔️ Expected result

Undoing works just fine

### ❌ Actual result

Editor crashes:
![table_undo2](https://user-images.githubusercontent.com/34380544/75984684-25a33580-5eeb-11ea-978a-c99b04747e98.gif)

### Errors:
```js
Uncaught CKEditorError: model-selection-range-intersects: Trying to add a range that intersects with another range in the selection. Read more: https://ckeditor.com/docs/ckeditor5/latest/framework/guides/support/error-codes.html#error-model-selection-range-intersects
 {"addedRange":{"start":{"root":"main","path":[1,1,0],"stickiness":"toNext"},"end":{"root":"main","path":[1,1,1],"stickiness":"toPrevious"}},"intersectingRange":{"start":{"root":"main","path":[1,1,0],"stickiness":"toNext"},"end":{"root":"main","path":[1,1,1],"stickiness":"toPrevious"}}}
    at LiveSelection._checkRange (http://localhost:8125/ckeditor5-table/tests/manual/tableselection.js:64846:11)
    at LiveSelection._prepareRange (http://localhost:8125/ckeditor5-table/tests/manual/tableselection.js:51700:8)
    at LiveSelection._fixGraveyardSelection (http://localhost:8125/ckeditor5-table/tests/manual/tableselection.js:52022:26)
    at Model.LiveSelection.listenTo.priority (http://localhost:8125/ckeditor5-table/tests/manual/tableselection.js:51514:10)
    at Model.fire (http://localhost:8125/ckeditor5-table/tests/manual/tableselection.js:124433:30)
    at Model.<computed> [as applyOperation] (http://localhost:8125/ckeditor5-table/tests/manual/tableselection.js:126570:16)
    at UndoCommand._undo (http://localhost:8125/ckeditor5-table/tests/manual/tableselection.js:119730:11)
    at Object.callback (http://localhost:8125/ckeditor5-table/tests/manual/tableselection.js:120017:9)
    at Model._runPendingChanges (http://localhost:8125/ckeditor5-table/tests/manual/tableselection.js:54477:58)
    at Model.enqueueChange (http://localhost:8125/ckeditor5-table/tests/manual/tableselection.js:53903:10)
```

```js
Uncaught TypeError: Cannot read property 'getChildIndex' of null
    at TableUtils.getCellLocation (tableutils.js:59)
    at TableSelection.getSelectedTableCells (tableselection.js:214)
    at getSelectedTableCells.next (<anonymous>)
    at TableSelection._updateModelSelection (tableselection.js:265)
    at TableSelection.stopSelection (tableselection.js:180)
    at MouseSelectionHandler._handleMouseMove (mouseselectionhandler.js:99)
    at Document.<anonymous> (mouseselectionhandler.js:64)
    at Document.fire (emittermixin.js:209)
    at MouseEventsObserver.fire (domeventobserver.js:96)
    at MouseEventsObserver.onDomEvent (mouseeventsobserver.js:42)
```

## 📃 Other details

* Browser: any
* OS: any

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by Reinmar** [comment]
Could you paste the stack trace? It's very helpful when looking for similar issues. Often, similar issues give the same error and we can easily find duplicates this way.

**Comment by FilipTokarski** [comment]
Wow sorry for that, I was sure I added the errors, must have missed something 🤦‍♂ Anyway, I updated the report.

**Comment by Reinmar** [comment]
There's a workaround in some remove col/row commands so they select the entire table before starting operating on cells. This works around the same issue – document selection is not able to properly manage its ranges when the document changes (cells are removed, ranges get insersected).

**Comment by Reinmar** [comment]
May be resolved by #6501.
